src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -12,7 +12,7 @@ namespace Microsoft.CodeAnalysis.Completion
     /// One of many possible completions used to form the completion list presented to the user.
     /// </summary>
     [DebuggerDisplay("{DisplayText}")]
-    public sealed class CompletionItem : IComparable<CompletionItem>
+    public class CompletionItem : IComparable<CompletionItem>
```

Comment:
def make sure this goes through some design review.

Reply:
I have removed the new subtype and reverted those changes to public API

Reply:
From our offline discussion:

Using `CompletionItem` causes additional overhead for the first run. From the traces I collected, the thread-time is bumped from 260ms to 320ms for creating 20k items, a big part of which is creating ImmutableDictionaries (~20ms) and GC (~40ms, the chance of which increases with additional allocations).

However, after cache is populated, the cost reduced to ~8ms for each call, identical to using a subtype, which is expected.

As a result, we decided to take a perf hit for initial run for a cleaner implementation.